reason-parser/src/reason_pprint_ast.ml

```diff
@@ -3364,39 +3437,60 @@ class printer  ()= object(self:'self)
      token "+", in `x + a * b` should not reduce until after the a * b gets
      a chance to reduce. This function would determine the minimum parens to
      ensure that. *)
-  method ensureContainingRule ~withPrecedence ~reducesAfterRight =
+  method ensureContainingRule ~withPrecedence ~reducesAfterRight () =
     match self#unparseExprRecurse reducesAfterRight with
     | (SpecificInfixPrecedence ({reducePrecedence; shiftPrecedence}, rightRecurse)) ->
-      if higherPrecedenceThan shiftPrecedence withPrecedence then rightRecurse
+      if higherPrecedenceThan shiftPrecedence withPrecedence then begin 
+        Simple rightRecurse
```

Comment:
I think this is a misnomer to wrap in `Simple`. This function `ensureContainingRule` ensures that `reducesAfterRight` will reduce *before* the containing rule with `withPrecedence` reduces. I realize that's pretty confusing and I have to read that sentence several times to make sense of it.

I think a way I could have wrir way to write the function definition to make it more clear would have been:

```
method ensureContainingRule ~withPrecedence:precedence ~reducesAfterRight:rightExpr () =
```

But the point of the function is to ensure that `rightExpr` will reduce at the proper time when it is reparsed, possibly wrapping it in parenthesis if needed. But that doesn't mean it is necessarily *simple*. For example, in `4 + 5*3`, `5*3` will reduce before the `+` operation, yet `5*3` is not "simple". Simple means it is clearly one token (such as `(anything)` or `[anything]` or `identifier`).